examples/07_join_aggregation.py

```diff
@@ -0,0 +1,438 @@
+"""
+TODO : Add some docstring
+----
+"""
+#!/usr/bin/env python
+# coding: utf-8
+
+# # Movies recommendation with explicit labels
+
+# ## Loading data
+
+# In[1]:
+
+
+from skrub.datasets import fetch_movielens
+
+ratings = fetch_movielens(dataset_id="ratings")
+ratings = ratings.X
+ratings
+
+
+# In[2]:
+
+
+import pandas as pd
+
+ratings["timestamp"] = pd.to_datetime(ratings["timestamp"], unit="s")
+ratings["year"] = ratings["timestamp"].dt.year
+ratings["day_name"] = ratings["timestamp"].dt.day_name()
+
+
+# ## Quick Data Exploration
+
+# In[3]:
+
+
+from matplotlib import pyplot as plt
+import seaborn as sns
+
+sns.set_style("darkgrid")
+
+
+def make_barplot(x, y, title):
+    norm = plt.Normalize(y.min(), y.max())
+    cmap = plt.get_cmap("magma")
+
+    ax = sns.barplot(
+        x=x,
+        y=y,
+        palette=cmap(norm(y)),
+    )
+    plt.xticks(rotation=30)
+    plt.xlabel(None)
+    plt.ylabel(None)
+    plt.tight_layout()
+    plt.title(title)
+
+
+# In[4]:
+
+
+year_volume = ratings["year"].value_counts().sort_index()
+
+make_barplot(
+    x=year_volume.index,
+    y=year_volume.values,
+    title="Yearly volume of ratings",
+)
+
+
+# In[5]:
+
+
+ratings["day_index"] = ratings["timestamp"].dt.weekday
+
+day_name_index = (
+    ratings[["day_index", "day_name"]].drop_duplicates().set_index("day_index")
+)
+day_name_index.sort_index()
+
+
+# In[6]:
+
+
+daily_volume = (
+    ratings["day_index"].value_counts().sort_index().to_frame().join(day_name_index)
+)
+
+make_barplot(
+    x=daily_volume["day_name"], y=daily_volume["count"], title="Daily volume of ratings"
+)
+
```

Comment:
I think having the previous plot on the years is much more interesting and revealing time/n_ratings relationship than this one on individual days. For simplification and shortening purpose, I think removing this and going directly the next plot would not hurt.